aeppl/truncation.py

```diff
@@ -0,0 +1,126 @@
+import warnings
+from typing import List, Optional
+
+import aesara.tensor as at
+import numpy as np
+from aesara.assert_op import Assert
+from aesara.graph.basic import Node
+from aesara.graph.fg import FunctionGraph
+from aesara.graph.opt import local_optimizer
+from aesara.scalar.basic import Clip
+from aesara.tensor.elemwise import Elemwise
+from aesara.tensor.random.op import RandomVariable
+from aesara.tensor.var import TensorConstant
+
+from aeppl.logprob import _logcdf, _logprob
+
+
+# TODO: Add interval transform
+class CensoredRV(RandomVariable):
+    r"""A base class for censored `RandomVariable`\s."""
+
+    def __init__(self, *args, base_op, **kwargs):
+        self.base_op = base_op
+        super().__init__(*args, **kwargs)
+
+
+@_logprob.register(CensoredRV)
+def censor_logprob(op, value, *inputs, name=None, **kwargs):
+
+    *rv_params, lower_bound, upper_bound = inputs
+    logprob = _logprob(op.base_op, value, *rv_params, name=name, **kwargs)
```

Comment:
You could alternatively leave the original `RandomVariable` in the graph, but make in an input of a `CensoredRV`, and this log-probability term would be taken care of automatically, then you can use this dispatch function to add the extra log-probability terms introduced by censoring.

Reply:
In that case I will have to direct the `value` var to the original variable in the rewrite phase (while keeping it pointing to the `censor_logprob` of course). I can give it a try, but I am not sure it will result in simpler code.

Reply:
After #19 is merged, you could also use `.tag.ignore_logprob`.  With that, we can write a general `RandomVariable` wrapper/view `Op` that can be used as the base class for something like `CensoredRV`.